hacep/src/main/java/org/kie/hacep/core/infra/DefaultSessionSnapShooter.java

```diff
@@ -103,23 +93,28 @@ public SnapshotInfos deserialize() {
             }
             consumer.close();
             SnapshotMessage snapshotMsg = bytes != null ? SerializationUtil.deserialize(bytes) : null;
-
+            /* if the Snapshot is ok we use the KieContainer and KieSession from the infos,
+            otherwise (empty system) we will create kiecontainer and kieSession from the envConfig's GAV */
+            KieContainer kieContainer = null;
             if (snapshotMsg != null) {
                 try (ByteArrayInputStream in = new ByteArrayInputStream(snapshotMsg.getSerializedSession())) {
-                    KieSessionConfiguration conf = KieServices.get().newKieSessionConfiguration();
+
+                    KieSessionConfiguration conf = srv.newKieSessionConfiguration();
                     conf.setOption(ClockTypeOption.get("pseudo"));
-                    kSession = marshallers.newMarshaller(kieContainer.getKieBase()).unmarshall(in,
-                                                                                               conf,
-                                                                                               null);
+                    String[] partsSerialized = snapshotMsg.getKjarGAV().split(":");
```

Comment:
This could be null if the snapshot has been created in a not updatable configuration

Reply:
Fixed with the centralized KieContainerUtils.getKieContainer